### ISSUE TITLE
Rethink of directory structure to make update easy

### ISSUE THREAD

**Issue by ki-foobar** [elonafoobar/elonafoobar#1221]
# Current directory structure

```
Elona foobar    The executable
profile/
  _/    The special profile that all profiles are created by. It is immutable.
    mod/
      core/
      kiroku/
        (*)
  profile_a/    Your owned profile
    mod/    Mod folder
      core/
        data/
        locale/
        graphic/
        and so on...
      kiroku/
      any_user_created_mod/
    save/    Save folder
      save_1/
      save_2/
    user/
```

> (*) `kiroku` mod should be removed at some point.

Due to the structure, it is hard to update Elona foobar. When you updated foobar,

1. Download the latest release and extend the zip file.
2. Copy all profiles under `profile` folder except `profile/_`.
3. Also copy `config.hcl`. The new version can add new options, but they should have their own default values.
4. Replace `mod/core` for each profile with `profille/_/mod/core` bundled with the new version.

Step 1-3 is necessary regardless of directory structure, but step 4 can be made easy by a wise structure.

# Proposal

```
Elona foobar    The executable
profile/
  profile_a/    Your owned profile
    mod/    Mod folder
      core/
        mod.hcl   This file specifies the version of the mod.
        config.hcl
        keybindings.hcl
        Any other profile-local resources
      any_user_created_mod/
    save/    Save folder
      save_1/
      save_2/
    user/
mod/
  core-1.0.0/
  core-1.1.0/
  core-2.0.1/
  any_user_created_mod-1.0.0/
  any_user_created_mod-1.1.0/
  (*)
```

> (*) These mods folder does *not* have any persistent data. Such data is saved in `profile/{profile_name}/mod/{mod_name_without_version}`

The update phase is as follows:

1. Download the latest release and extend the zip file.
2. Copy all profiles under `profile` folder.
3. Also copy `config.hcl`.
4. Copy `mod` folder. If `core` mod has been changed in the new version, it should have different version number and folder name. Thus, old one and new one will not conflict with each other.

At step 4, you have to copy only `mod` folder. The current directory structure requires you to copy `mod/core` several times (the number of profiles you have).

In addition, this method gives us another good point. It is that downloading & installing mods cost less time because the same mod is installed only once.

**Comment by ki-foobar** [comment]
I noticed that the proposal is git-unfriendly as git holds snapshots and does not fully support renaming/moving files (such operations are recorded as "delete and add").

**Comment by ki-foobar** [comment]
The proposal is git-unfriendly as git holds snapshot and does not fully support renaming/moving files. Thus, it would be better to remain the name of `mod/core` folder in git repository (not to rename it to `mod/core-{version identifier}`) and rename it to the correct name in `bin` folder which is git-ignored.

**Comment by ki-foobar** [comment]
It was closed by #1263.
